### PR TITLE
Kong API Gateway Implementation

### PR DESCRIPTION
**Pull request by pshumateEDC**
Simple Kong implementation. Deployable to cloud.gov using Cloud Formation. Deployed with a basic api key authentication plugin. Unit tests included for ensuring plugin is deployed properly.

### CODE REVIEW

File: gitignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+tests/__pycache__/
```
Comment (dzaslavskiy) [inline gitignore:1]: should be '.gitignore'

Comment (pshumateEDC) [reply gitignore]: Done

File: post-deployment.sh [hunk 2]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+echo 'post deployment script'
+cf ssh -N -T -L 8081:localhost:8081 give-api-gateway &
+PID=$!
+sleep 3
+curl -X POST http://localhost:8081/plugins/ --data "name=key-auth"
+pytest tests/KongTest.py
```
Comment (dzaslavskiy) [inline post-deployment.sh:7]: renamed file

Comment (pshumateEDC) [reply post-deployment.sh]: File name changed

**Review by dzaslavskiy — APPROVED** [review]

**Pull request merged**